### PR TITLE
[tests-only]Added tests for folder creation and syncing (part 2)

### PR DESCRIPTION
**Pull request by Talank**
### Description
Added automated tests for folder creation and syncing

### Related Issues
- Part of https://github.com/owncloud/client/issues/8971#issuecomment-909884257

### CODE REVIEW

File: test/gui/tst_syncing/test.feature [hunk 1]
```diff
@@ -256,3 +256,34 @@ Feature: Syncing files
...
+        And as "Alice" folder "<foldername>/<foldername>" should exist on the server
+        And as "Alice" folder "<foldername>/<foldername>/<foldername>" should exist on the server
+        And as "Alice" folder "<foldername>/<foldername>/<foldername>/<foldername>" should exist on the server
+        And as "Alice" folder "<foldername>/<foldername>/<foldername>/<foldername>/<foldername>" should exist on the server
+        Examples:
+            | foldername                                                      |
+            | An empty folder which name is obviously more than 59 characters |
+
+    
+    Scenario: Invalid system names are synced in linux
```
Comment (Talank) [inline test/gui/tst_syncing/test.feature:279]: We can remove this scenario if we include it in the scenarios `Syncing a file to the server` and `Syncing a folder to the server`

File: test/gui/tst_syncing/test.feature [hunk 2]
```diff
@@ -256,3 +256,34 @@ Feature: Syncing files
...
+        Given user "Alice" has set up a client with default settings
+        And user "Alice" has created folder "COM" on the server
+        And user "Alice" has created folder "test%" on the server
+        And user "Alice" has uploaded file on the server with content "server content" to "/PRN"
+        And user "Alice" has uploaded file on the server with content "server content" to "/foo%"
+        When the user waits for file "/foo%" to be synced
```
Comment (saw-jan) [inline test/gui/tst_syncing/test.feature:285]: You have created the files and folders in the server and asserted that they exist in the server.
Test for client?

File: test/gui/tst_syncing/test.feature [hunk 3]
```diff
@@ -256,3 +256,34 @@ Feature: Syncing files
...
+        Then as "Alice" folder "COM" should exist on the server
+        And as "Alice" folder "test%" should exist on the server
+        And as "Alice" file "/PRN" should exist on the server
+        And as "Alice" file "/foo%" should exist on the server
```
Comment (saw-jan) [inline test/gui/tst_syncing/test.feature:289]: You have created the files and folders in the server and asserted that they exist in the server.
Test for client?

File: test/gui/tst_syncing/test.feature [hunk 4]
```diff
@@ -257,3 +257,38 @@ Feature: Syncing files
...
+            | An empty folder which name is obviously more than 59 characters |
+
+    
+    Scenario: Invalid system names are synced in linux
+        Given user "Alice" has set up a client with default settings
+        And user "Alice" has created folder "COM" on the server
+        And user "Alice" has created folder "test%" on the server
+        And user "Alice" has uploaded file on the server with content "server content" to "/PRN"
+        And user "Alice" has uploaded file on the server with content "server content" to "/foo%"
+        When the user waits for file "/foo%" to be synced
```
Comment (saw-jan) [inline test/gui/tst_syncing/test.feature:286]: waiting only for one file? Other files might complain :smile: 
```suggestion
        When the user waits for the files to be synced
```

File: test/gui/tst_syncing/test.feature (hunk 1 above)
Comment (swoichha) [reply test/gui/tst_syncing/test.feature:280]: I think it is better this way.

**Review by swoichha — APPROVED** [review]
lgtm

**Review by saw-jan — APPROVED** [review]
LGTM

**Review by saw-jan — APPROVED** [review]
LGTM

**Review by saw-jan — APPROVED** [review]
LGTM

**Pull request merged**